packages/core-mobile/app/screens/rpc/components/v2/ApprovalPopup.tsx

```diff
@@ -112,6 +109,25 @@ const ApprovalPopup = (): JSX.Element => {
     }
   }, [rejectAndClose, request, account])
 
+  const filteredSections = useMemo(() => {
+    return displayData.details.map(detailSection => {
+      if (detailSection.title !== 'Transaction Details') {
+        return detailSection
+      }
+
+      const filteredItems = detailSection.items.filter(item => {
+        if (typeof item === 'string') return true
+
+        const isDataOrInAppWebsite =
+          item.label === 'Website' && isInAppRequest(request)
+
+        return !isDataOrInAppWebsite
+      })
+
+      return { ...detailSection, items: filteredItems }
+    })
+  }, [displayData.details, request])
```

Comment:
I think this logic should live in the DetailSection component. you can pass a boolean prop to it for isInAppRequest.

Reply:
I want to keep the DetailSectionView component as simple as possible. If we decide to hide certain sections based on specific logic, I believe that logic should be applied at a higher level rather than within DetailSectionView. This approach will make the code’s intent clearer.